### PR TITLE
Allow user to send and recieve custom messages to and from the SMT solver

### LINKED ISSUE #30: Support for sending custom expressions to the solver
Hi,

`easy_smt` currently provides nice access to creating custom `atom`s and using them in custom expressions, but has no way to actually send these to the solver for the user (as `solver.send` is private). This is useful for example with [optimization in z3](https://microsoft.github.io/z3guide/docs/optimization/arithmeticaloptimization/) which provides special `(minimize expr)` and `(maximize expr)` expressions that need to be sent to the solver.

Such expressions can be made using the following
```rust
let minimize = self.sol.atom("minimize");
let expr = self.sol.list(vec![minimize, self.minimize_expr.clone()]);

self.sol.send(expr); // missing
```
but they cannot yet be sent to the solver. This should be a relatively simple patch to include a wrapper for `solver.send`.

### PR DESCRIPTION
**Pull request by UnsignedByte**
Fixes #30

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -176,6 +176,24 @@ pub struct Context {
...
+    }
+
+    /// Directly receive a response from the solver.
+    pub fn recv(&mut self) -> io::Result<SExpr> {
+        let solver = self
+            .solver
+            .as_mut()
+            .expect("recv requires a running solver");
+        solver.recv(&self.arena)
+    }
```
Comment (fitzgen) [inline src/context.rs:195]: Mind adding a `raw_` prefix (or something, open to bikeshedding) and a note in the doc comment to signify that these are low-level, "raw" APIs, `check`/`set_option`/etc. should generally be preferred, and are only intended as an escape hatch for solver-specific commands that this crate doesn't have built-in support for?

**Review by fitzgen — COMMENTED** [review]
Thanks!

File: src/context.rs (hunk 1 above)
Comment (UnsignedByte) [reply src/context.rs]: Done! I also moved these calls to the bottom of the `impl` just so that they aren't at the top of `cargo doc`.

**Review by fitzgen — APPROVED** [review]
Thanks!

**Comment by fitzgen** [comment]
Published 0.2.3 with this

**Pull request merged**